### ISSUE TITLE
Manuskript crashing while booting

### ISSUE THREAD

**Issue by Inthesouth** [olivierkes/manuskript#1021]
Hi, I've been using Manuskript on its last version (0.13.1) for the last couple of weeks. Everything seemed to work fine until the other day I tried to run Manuskript as usual, but it crashed while booting. After a process to figure out what the heck was going on, I found out that my .msk project was causing the whole mess. It has custom name, in this case "Escritura 2022" which it means "2022 Writing", but I don't think this could be the problem. The file is 27,5 KB, so I hope to be able to recover my work.

When I open the .exe, this is the error reported:

An unhandled exception has occurred!
Traceback (most recent call last):
  File "manuskript\ui\welcome.py", line 165, in openFile
  File "manuskript\mainWindow.py", line 614, in loadProject
  File "manuskript\mainWindow.py", line 900, in loadDatas
  File "manuskript\loadSave.py", line 60, in loadProject
ValueError: invalid literal for int() with base 10: '\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x0
CRITICAL> An unhandled exception has occurred!
Traceback (most recent call last):
  File "manuskript\ui\welcome.py", line 165, in openFile
  File "manuskript\mainWindow.py", line 614, in loadProject
  File "manuskript\mainWindow.py", line 900, in loadDatas
  File "manuskript\loadSave.py", line 60, in loadProject
ValueError: invalid literal for int() with base 10: '\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x0

I hope you can help me. I'm pretty desperate right now. Thanks in advance.

P.S. I tried the "Manuskript Crashes on Project Open #741" submitted by StacyWV, but it didn't work out.

**Comment by TheJackiMonster** [comment]
It looks like the application can't open the project file. I assume it is still an archive from the save-as-single-file mode which isn't recommended to use. You can try to open the files with another application to extract .zip archives since our single-file projects are just using typical zip compression.

If that doesn't help, it will probably not be possible to recover much. So I hope you have at least a backup to roll back to. It would still be interesting why the project gets corrupted at any point to solve an issue like this from happening.

**Comment by Inthesouth** [comment]
Hi. I appreciate your answer.

In my first comment I said that I tried the "Manuskript Crashes on Project Open https://github.com/olivierkes/manuskript/issues/741" submitted by StacyWV and it didn't work. This means that I'm not able to open as .zip my project because it keep saying that there is no data in it or it is corrupted. Not in Windows 11, not in Debian 11. But the file still is 27,5 kb. Additionally I opened with a Hex Editor and the file is full with 00, nothing else. At that point I assumed that my work was lost.

I didn't have any backup for two reasons:
1) I've only been using the software for the last month or so and I didn't expect problems so quickly.
2) I thought the software kept track of previous versions.
My bad, I suppose.

And yes, my file was an "archive from the save-as-single-file mode". I am not any kind of expert in software, but I try to support open source as much as I can. That's the reason I really didn't go deep into all settings, but for the ones related to customization the skin and so of the software. So, this single file mode I assume it was set as the default option.

To sum up, I assume my work is lost which is going to cause me some issues at job, but it is what it is.
Besides, I want to keep using Manuskript because I think it is a great software, but I want to be sure to this thing not happening all again. So, the safest option is the folder save. Ok. In case of crashing again, as the file is a folder, could it be more accessible or accessible at all?

Thank you.

**Comment by TheJackiMonster** [comment]
Hmm, I understand. I hope we can get rid of those problems with the single-file-mode in the future. I have actually written a cleaner code to replace it but because it's not easily compatible with the current functions and API, we can't simply put it in yet.

I recommend to use to save projects as folders. You can even use SVN or GIT on your projects then because the files describing your projects data are all text based and readable. So committing changes via GIT is quite efficient in many cases and you will have any version of your project to fall back (we even plan to replace our current revision system with something like that because the current one is quite broken and not recommended to use).

In any way you can also make copies of your projects after a lot of changes to backup data and replace files on a potential loss/crash to restore data. However if you save your projects as folders, it is likely that you will only loose one file (maybe a chapter, a character or an object...) at best. The save as one file mode makes any bug in the process to affect everything though... unfortunately.

**Comment by Inthesouth** [comment]
About "You can even use SVN or GIT", I'm not an expert but I'm interested in learning. Do you know any info, video, tutorial I can follow to implement it?

Thank you!

**Comment by obw** [comment]
SVN is like GIT a Tool which is manly used by developers tor keep track over the changes of their Source Code. But they could also be used for Manuskript-Projects, I personally use SVN, because I use it over fifteen Years, and I'm not so used to GIT.

The other benefit for e is, that it's easier to set up a SVN Server, then a GIT-Server. So my Editor can easily access the latest Changes, and also I'm able to get his changes easily.

You do not need a Server to run one of both Systems, the both support file based repositories!

The best resource for learning SVN is this page: https://svnbook.red-bean.com/

Sorry but for GIT I can't help.

I think there are a lot of tutorials on YT, but I don't use them, so I can't give you an example.

Regards

**Comment by TheJackiMonster** [comment]
I assume it depends which operating system you use to pick an interface to manage/use SVN or GIT. I personally use [gitg](https://wiki.gnome.org/Apps/Gitg) on Linux but there also seem to be Windows binaries. The advantage of it is that it pretty much allows most features of git via GUI like adding a project or an empty directory and creating a repository for it automatically.

As said by obw you can use GIT and SVN without a server. Using a server is only useful to distribute your projects or exchange/merge progress between multiple devices. GIT is especially designed to handle merges of different versions from people written in parallel and you can also exchange changes via mail but I would recommend looking into command line to do stuff like that.

Using Gitg would be like this:

1. You add a project to create a repository for it or select it on startup from the list.
2. You switch to the changes tab (via right button at the top showing an arrow pointing downwards).
3. You add/stage your changed files (new files will be "untracked" by default meaning ignored for commits, previously committed files will be in "unstaged" by default).
4. Once you staged all files, you commit those changes with a custom message (using the button right at the bottom).
5. The commit should show up in the commit history view (otherwise you can reload via a button in the options menu).

So like this you would make commits of a state from your project. Most at the time you don't need to do anything else than committing changes. If anything goes wrong or you want to roll back to any previous state:

1. Select the repository from the list on startup.
2. Select the commit to go back to.
3. Right-click on it to "create branch" (the name of the branch doesn't matter but you can pick something fitting)
4. It should be checked "checkout the created branch" by default but otherwise you can check out any branch listed on the left side under "branches" with a double-click or right-click and "checkout".

Checking out a branch or commit (which can be done from command line directly at least) will automatically switch your files to that specific committed state. The only thing you should remember is committing all unstaged and untracked changes before switching branch just to be sure, you can definitely go back (the only exception would be a totally corrupted state which you don't want to go back to).

All commits in GIT don't have to be sequential by the way. So you can fork at any point in your history and theoretically merge different paths later on (but this might require manual merge processing which I wouldn't recommend at the beginning of learning GIT).

**Comment by Inthesouth** [comment]
I'm going to check it. Thank you.
